src/librustc/ty/util.rs

```diff
@@ -667,7 +667,8 @@ impl<'a, 'gcx, 'tcx, W> TypeVisitor<'tcx> for TypeIdHasher<'a, 'gcx, 'tcx, W>
             }
             TyProjection(ref data) => {
                 self.def_id(data.trait_ref.def_id);
-                self.hash(data.item_name.as_str());
+                // FIXME(tschottdorf): are there consequences to this change?
```

Comment:
we don't actually need the first hash (of `data.trait_ref.def_id`); the second should suffice, I think.